### ISSUE TITLE
NSOperationQueue is canceled it will Clash.

### ISSUE THREAD

**Issue by yoneapp** [daltoniam/SwiftHTTP#203]
Hi.
NSOperationQueue is canceled it will Clash.

Sample Code:

```swift
import UIKit
import SwiftHTTP

class ViewController: UIViewController {
    
    let operationQueue = NSOperationQueue()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        operationQueue.maxConcurrentOperationCount = 2
        
        do {
            for _ in 1...100 {
                let opt = try HTTP.New("https://google.com", method: .GET)
                opt.onFinish = { response in
                    //do stuff
                }
                operationQueue.addOperation(opt)
            }
            
        } catch let error {
            print("got an error creating the request: \(error)")
        }
        
        let delay = 1 * Double(NSEC_PER_SEC)
        let time  = dispatch_time(DISPATCH_TIME_NOW, Int64(delay))
        
        dispatch_after(time, dispatch_get_main_queue(), {
            self.operationQueue.operations.forEach { $0.cancel() }
        })
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
}
```

Logs:

```
2015-12-12 10:45:10.068 SwiftHTTPOperationBug[25326:714091] *** SwiftHTTP.HTTP 0x7a1ecd90 went isFinished=YES without being started by the queue it is in
2015-12-12 10:45:10.068 SwiftHTTPOperationBug[25326:714091] *** SwiftHTTP.HTTP 0x7a06b880 went isFinished=YES without being started by the queue it is in
2015-12-12 10:45:10.069 SwiftHTTPOperationBug[25326:714091] *** SwiftHTTP.HTTP 0x7a072240 went isFinished=YES without being started by the queue it is in
2015-12-12 10:45:10.069 SwiftHTTPOperationBug[25326:714091] *** SwiftHTTP.HTTP 0x7a06e340 went isFinished=YES without being started by the queue it is in
2015-12-12 10:45:10.069 SwiftHTTPOperationBug[25326:714091] *** SwiftHTTP.HTTP 0x7a071d10 went isFinished=YES without being started by the queue it is in
2015-12-12 10:45:10.070 SwiftHTTPOperationBug[25326:714091] *** SwiftHTTP.HTTP 0x7a082ec0 went isFinished=YES without being started by the queue it is in
2015-12-12 10:45:10.070 SwiftHTTPOperationBug[25326:714091] *** SwiftHTTP.HTTP 0x7a0835d0 went isFinished=YES without being started by the queue it is in
2015-12-12 10:45:10.070 SwiftHTTPOperationBug[25326:714091] *** SwiftHTTP.HTTP 0x7a083ce0 went isFinished=YES without being started by the queue it is in
2015-12-12 10:45:10.070 SwiftHTTPOperationBug[25326:714091] *** SwiftHTTP.HTTP 0x7a0843f0 went isFinished=YES without being started by the queue it is in
2015-12-12 10:45:10.070 SwiftHTTPOperationBug[25326:714091] *** SwiftHTTP.HTTP 0x7a084b00 went isFinished=YES without being started by the queue it is in
```

And the app is cache.

Thanks.

**Comment by yoneapp** [comment]
It was fixed this.
I will send a pull request.

**Comment by daltoniam** [comment]
fixed by #204
